worlds/trackmania/items.py

```diff
@@ -0,0 +1,159 @@
+
+    if spots_remaining < 0:
+        spots_remaining = 0 # just in case
+
+    skip_count = min(round(float(spots_remaining) * (world.options.skip_percentage / 100.0)),spots_remaining)
+    itempool += create_items(world, "Map Skip", skip_count)
+
+    discount_count = min(round(float(spots_remaining) * (world.options.discount_percentage / 100.0)),spots_remaining-skip_count)
+    itempool += create_items(world, "PB Discount", discount_count)
+
+    filler_count = spots_remaining - skip_count - discount_count
+    for x in range(filler_count):
+        filler_name = get_filler_item_name(world)
+        itempool += create_items(world,filler_name, 1)
+
+    return itempool
+
+def create_medals(world: "TrackmaniaWorld", medal: str, minimum_target_time:int, map_count: int) -> list[Item]:
+    if get_progression_medal(world) != medal and not get_medal_enabled(world, medal):
+        return []
+    if world.options.target_time >= minimum_target_time:
+        return create_items(world, medal, map_count)
+    return []
+
+def create_item(world: "TrackmaniaWorld", name: str) -> Item:
+    item_id = world.item_name_to_id[name]
+    return TrackmaniaItem(name, determine_item_classification(world,name), item_id, world.player)
+
+def create_items(world: "TrackmaniaWorld", name: str, count: int) -> list[Item]:
+
+    item_id = world.item_name_to_id[name]
+    itemlist: list[Item] = []
+
+    for i in range(count):
+        itemlist += [TrackmaniaItem(name, determine_item_classification(world,name), item_id, world.player)]
+
+    return itemlist
+
+def get_filler_item_name(world: "TrackmaniaWorld") -> str:
+    return filler_item_names[world.random.randint(0, len(filler_item_names)-1)]
+
+def get_medal_enabled(world: "TrackmaniaWorld", medal: str) -> bool:
+    match medal:
+        case "Bronze Medal":
+            return world.options.disable_bronze_medals <= 0
+        case "Silver Medal":
+            return world.options.disable_silver_medals <= 0
+        case "Gold Medal":
+            return world.options.disable_gold_medals <= 0
+        case "Author Medal":
+            return True
+        case _:
+            return True
+
+def get_locations_per_map(world: "TrackmaniaWorld") -> int:
+    checks: int = 1
+    if world.options.target_time < 100 or world.options.disable_bronze_locations <= 0:
+        checks += 1
+    if world.options.target_time >=100 and world.options.disable_silver_locations <= 0:
+        checks += 1
+    if world.options.target_time >=200 and world.options.disable_gold_locations <= 0:
+        checks += 1
+    if world.options.target_time >=300 and world.options.disable_author_locations <= 0:
+        checks += 1
+    return checks
+    
+
+def build_items() -> dict[str,int]:
+    # hardcoded items
+    items: dict[str, int] = trackmania_items
+
+    # filler items
+    filler_id:int = base_filler_id
+    for x in range(len(filler_item_names)):
+        items[filler_item_names[x]] = filler_id
+        filler_id += 1
```

Comment:
```suggestion
    for x, name in enumerate(filler_item_names):
        items[name] = base_filler_id + x
```
I think this is much more succinct